### PR TITLE
releng: Update Slack branch warnings

### PR DESCRIPTION
**Pull request by justaugustus**
- Move branch alerts to #release-management
- Remove RelMgr exemptions in kubernetes/kubernetes Slack warnings

  We don't actually require exemption anymore, as Release Managers no longer fast-forward branches

- Remove Slack exemptions from k/release `prototype` branch 

Continuation of https://github.com/kubernetes/release/pull/2042 and discussion in https://kubernetes.slack.com/archives/C2C40FMNF/p1620060717262100.

I would've liked to exempt dependabot from triggering Slack warnings, but it's an app, not a GitHub user, and I'm unsure how that will behave w/ the merge warning Prow plugin.
EDIT: Instead, we'll compromise and move the alerts to the #release-management channel.

cc: @kubernetes/release-engineering

### CODE REVIEW

**Review by hasheddan — APPROVED** [review]
/lgtm

File: config/prow/plugins.yaml
Comment (cblecker) [inline config/prow/plugins.yaml:271]: This would exempt these users from all branches, including master/default. Why would we want this?

**Review by cblecker — CHANGES_REQUESTED** [review]
/hold

File: config/prow/plugins.yaml
Comment (justaugustus) [reply config/prow/plugins.yaml]: @cblecker -- Thanks for the review!

That actually got me thinking...
We don't actually require exemption anymore, as Release Managers no longer fast-forward branches. 🙃 

I've dropped all of the Release Manager-specific branch exemptions.
/hold cancel

**Comment by cblecker** [comment]
/lgtm
/approve

🎉

**Pull request merged**
